### PR TITLE
OpenConfig/SONiC: add support for tcp-established option

### PR DESCRIPTION
**Pull request by XioNoX**
The OpenConfig implementation is experimental as only implemented based on the [OpenConfig spec](https://github.com/openconfig/public/commit/2de10a479bd0f537109108fa12bc3790aac01548) (not tested on a real device).
The SONiC implementation differs slightly from the OpenConfig one, thus the new generator inheriting from OpenConfig. This one has been tested on a real Dell SONiC device.

Fixes https://github.com/aerleon/aerleon/issues/306 and should be a decent baseline to support more options both OpenConfig and SONiC.

### CODE REVIEW

**Comment by ankenyr** [comment]
What do you mean here by "experimental"? Is this not in sync with the current openconfig specification? I am not sure we would want to accept this into the openconfig generator if it breaks things for others.

File: aerleon/lib/openconfig.py [hunk 1]
```diff
@@ -148,6 +163,17 @@ def ConvertToDict(
             protos = ['none']
 
         ace_dict = copy.deepcopy(term_dict)
+
+        # options
+        if self.term.option:
+            for opt in [str(x) for x in self.term.option]:
```
Comment (ankenyr) [inline aerleon/lib/openconfig.py:169]: Are there cases when options are not a string?

Comment (ankenyr) [inline aerleon/lib/openconfig.py:169]: Could this be simplified to `if self.term.protocol != ['tcp'] and 'tcp-established' in self.term.option `?
This should short circuit in the cases where the first option is false which is quick and with my other comment I don't believe options are not strings.

Comment (XioNoX) [reply aerleon/lib/openconfig.py:169]: Indeed, looks like not. I copied the logic from the Juniper's generator.

Comment (XioNoX) [reply aerleon/lib/openconfig.py:169]: I initially copied the code from the Juniper generator, the idea was that right now we only have 1 option, but the code needs to be ready to support more options. That said I don't think itterating over all the "options" are useful, but checking if a given option is in the array is enough (and cleaner). Let me know what you think of this new version.

**Comment by XioNoX** [comment]
I updated the tests to reach 100% of code coverage.

There is something wrong going on but not sure how to fix it. It's what being caught by the tests.

If I understand correctly, the `"destination-address": "10.2.3.4/32",` is being overwritten by `"protocol": 6` instead of `"protocol": 6` to be added alongside.

Any idea what could be going on?

File: tests/regression/openconfig/openconfig_test.py [hunk 2]
```diff
@@ -588,6 +604,32 @@ def testMixedDaddr(self):
         self.naming.GetNetAddr.assert_called_once_with('CORP_EXTERNAL')
         print(acl)
 
+    @capture.stdout
+    def testTcpEstablished(self):
+        self.naming.GetServiceByProto.return_value = ['80']
```
Comment (jtwb) [inline tests/regression/openconfig/openconfig_test.py:609]: @XioNoX I think you might have previously set a return value for GetNetAddr in this test but later removed it. I think you need to add it back. This suggested change fixes the test.

```suggestion
        self.naming.GetServiceByProto.return_value = ['80']
        self.naming.GetNetAddr.return_value = TEST_IPS
```

Comment (XioNoX) [reply tests/regression/openconfig/openconfig_test.py:609]: Thanks! Looks like that was it. I was looking for issues in the generator and not in the test.

**Review by ankenyr — APPROVED** [review]

**Pull request merged**